### ISSUE TITLE
QR Code not showing

### ISSUE THREAD

**Issue by jandreaucodes** [Ghawken/HomeKitLink-Siri#6]
**Describe the bug**
When I click "View QR Code", I just get a blank page with the Indigo branding on it.

**To Reproduce**
1. Click View QR Code
2. Browser launches
3. Page is blank, other than Indigo branding

**Expected behavior**
QR code is visible.

**Comment by Ghawken** [comment]
The bridge needs to be running for the QR code to show.
So you need to save the Config dialog and then reopen bridge to see QRCode
Does that resolve?

Otherwise there is probably a bigger issue here and debug logging etc will be needed

**Comment by jandreaucodes** [comment]
How do I get past "Starting up"?

It seems to get stuck on that step, and the bridge never fully starts.

**Comment by Ghawken** [comment]
Not sure that that is the case - does say starting up when done.  

But basically - need more info.

Turn debug logging on for everything - check all the debug boxes 1-9
Select Debugging logging.
Restart the plugin and post the output.

**Comment by jandreaucodes** [comment]
Here's all the logs from when I turned on debug until it stopped doing stuff:

````
HomeKitLink Siri Debug          logLevel = 5
   HomeKitLink Siri Debug          User prefs saved.
   HomeKitLink Siri Debug          Debugging on (Level: 5)
   Reloading plugin "HomeKitLink Siri 0.2.22" using API v3.0.0
   Stopping plugin "HomeKitLink Siri 0.2.22" (pid 37937)
   HomeKitLink Siri Debug          stop_concurrent_thread called
   HomeKitLink Siri                Completing full Bridge Shutdown ...
   HomeKitLink Siri Debug          stop_concurrent_thread called
   HomeKitLink Siri Debug          Siri: Stopping homeKitBridgeDevice Device 515860483
   HomeKitLink Siri Debug          Stopping Siri HomeKitLink Bridge with Unique Bridge ID 303271
   HomeKitLink Siri Debug          Checkid 303271 and total deviceBridgeNumber [303271]
   HomeKitLink Siri Debug          After: Checkid 303271 and total deviceBridgeNumber []
   HomeKitLink Siri Debug          Removed Bridge <accessory display_name='HomeKitLink Bridge 303271' services=['AccessoryInformation', 'HAPProtocolInformation']>
   HomeKitLink Siri Debug          Found Driver to stop Driver Id: 303271
   HomeKitLink Siri Debug          Driver to be deleted using port 51826 and portsinUse now set()
   HomeKitLink Siri Debug          Removed Driver <pyhap.accessory_driver.AccessoryDriver object at 0x103b767d0>
   HomeKitLink Siri                Siri has completed full Bridge, Driver and Bridge Thread Shutdown.
   Stopped plugin "HomeKitLink Siri 0.2.22"
   Starting plugin "HomeKitLink Siri 0.2.22" (pid 76688)
   HomeKitLink Siri Debug          logLevel = 5
   HomeKitLink Siri                
   HomeKitLink Siri                ================================================ Initializing New Plugin Session =================================================
   HomeKitLink Siri                Plugin name:                   HomeKitLink Siri
   HomeKitLink Siri                Plugin version:                0.2.22
   HomeKitLink Siri                Plugin ID:                     com.GlennNZ.indigoplugin.HomeKitLink-Siri
   HomeKitLink Siri                Indigo version:                2022.1.0
   HomeKitLink Siri                Python version:                3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
   HomeKitLink Siri                Python Directory:              /Library/Frameworks/Python.framework/Versions/3.10
   HomeKitLink Siri                
   HomeKitLink Siri                ================================================== End Initializing New Plugin  ==================================================
   Started plugin "HomeKitLink Siri 0.2.22"
   HomeKitLink Siri Debug          Starting Plugin. startup() method called.
   HomeKitLink Siri                Finding devices to publish to HomeKit Given Startup...
   HomeKitLink Siri Debug          Checking Plugin Prefs Directory
   HomeKitLink Siri Debug          Saving Camera Snapshots to:/Library/Application Support/Perceptive Automation/Indigo 2022.1/Preferences/Plugins/com.GlennNZ.indigoplugin.HomeKitLink-Siri/cameras
   HomeKitLink Siri Debug          create Device List Internal called
   HomeKitLink Siri Debug          Device: Sams Bed,  ID: 220544621  Model: Camera Type: SecuritySpyCamera  Sensor     ManufacturerName: org.cynic.indigo.securityspy
   HomeKitLink Siri Debug          Device: Sams Bed,  ID: 1866296567  Model: OutletLinc (2663) Type: SecuritySpyCamera  Sensor     ManufacturerName: 
   HomeKitLink Siri Debug          Siri: Starting homeKitBridgeDevice Device 515860483 
   HomeKitLink Siri Debug          checkid = 303271 and self.deviceBridgeNumber = []
   HomeKitLink Siri Debug          Starting Siri HomeKitLink Bridge with Unique Bridge ID 303271
   HomeKitLink Siri Debug          create Device List Internal called
   HomeKitLink Siri Debug          Device: Sams Bed,  ID: 220544621  Model: Camera Type: SecuritySpyCamera  Sensor     ManufacturerName: org.cynic.indigo.securityspy
   HomeKitLink Siri Debug          Device: Sams Bed,  ID: 1866296567  Model: OutletLinc (2663) Type: SecuritySpyCamera  Sensor     ManufacturerName: 
   HomeKitLink Siri Debug          Attempting to start a single Bridge:  Bridge Siri and ID 303271
   HomeKitLink Siri Debug          Next Port available:51826
   HomeKitLink Siri Debug          Asyncio created a new Event Loop
   HomeKitLink Siri Debug          Sets of Ports Currently in Use: {51826}
   HomeKitLink Siri Debug          set_value: Name to HomeKitLink Bridge 303271
   HomeKitLink Siri Debug          set_value: SerialNumber to default
   HomeKitLink Siri Debug          set_value: Version to 01.01.00
   HomeKitLink Siri Debug          get_bridge_multiple called and self.device_list_internal length 2
   HomeKitLink Siri Debug          Item :{'deviceid': 220544621, 'devicename': 'Sams Bed', 'accessory': None, 'aid': 220544621, 'subtype': 'SecuritySpyCamera', 'bridgeID': 303271, 'devicemodel': 'Camera', 'devicesensor': '', 'manufacturername': 'org.cynic.indigo.securityspy'} exisits for this bridge <accessory display_name='HomeKitLink Bridge 303271' services=['AccessoryInformation', 'HAPProtocolInformation']>
   HomeKitLink Siri Debug          Adding Accessory Name: Sams Bed,  ID  303271220544621 , Type: SecuritySpyCamera
   HomeKitLink Siri Debug          sorting out camera options for ID:220544621
   HomeKitLink Siri Debug          Startup Refresh Snapshots images to Default for Camera: /Library/Application Support/Perceptive Automation/Indigo 2022.1/Preferences/Plugins/com.GlennNZ.indigoplugin.HomeKitLink-Siri/cameras/BabyCam.jpg
   HomeKitLink Siri Debug          Config Now
{'support_audio': True, 'video_codec': 'libx264', 'audio_codec': 'libfdk_aac', 'max_height': 1520, 'max_width': 2688, 'max_fps': 10, 'video_map': '0:0', 'DoorBell_ID': '', 'SS_name': 'BabyCam', 'SS_serverip': '192.168.7.44', 'SS_serverport': '8000', 'SS_username': 'Indigo', 'SS_password': '**', 'SS_imageURL': 'http://Indigo:**@192.168.7.44:8000/++image?cameraNum=2', 'useMotionSensor': True, 'start_commands_extra': '-rtsp_transport tcp -probesize 64 -analyzeduration 0 ', 'stream_source': 'rtsp://Indigo:0623Megan0929@192.168.7.44:8000/stream?cameraNum=2&width=1280'}

   HomeKitLink Siri Debug          Options:

{'support_audio': True, 'video_codec': 'libx264', 'audio_codec': 'libfdk_aac', 'max_height': 1520, 'max_width': 2688, 'max_fps': 10, 'video_map': '0:0', 'DoorBell_ID': '', 'SS_name': 'BabyCam', 'SS_serverip': '192.168.7.44', 'SS_serverport': '8000', 'SS_username': 'Indigo', 'SS_password': '**', 'SS_imageURL': 'http://Indigo:**@192.168.7.44:8000/++image?cameraNum=2', 'useMotionSensor': True, 'start_commands_extra': '-rtsp_transport tcp -probesize 64 -analyzeduration 0 ', 'stream_source': 'rtsp://Indigo:0623Megan0929@192.168.7.44:8000/stream?cameraNum=2&width=1280'}

   HomeKitLink Siri Debug          set_value: Name to Sams Bed
   HomeKitLink Siri Debug          set_value: SerialNumber to default
   HomeKitLink Siri Debug          set_value: SupportedRTPConfiguration to AgEA
   HomeKitLink Siri Debug          set_value: SupportedVideoStreamConfiguration to AbwBAQACDwMBAAEBAQIBAAIBAQIBAgMMAQJAAQICtAADAgoAAwwBAkABAgLwAAMCCgADDAEC4AECAg4BAwIKAAMMAQLgAQICaAEDAgoAAwwBAoACAgJoAQMCCgADDAECgAICAuABAwIKAAMMAQIABAICQAIDAgoAAwwBAgAEAgIAAwMCCgADDAECAAUCAtACAwIKAAMMAQIABQICwAMDAgoAAwwBAoAHAgI4BAMCCgADDAECQAYCArAEAwIKAA==
   HomeKitLink Siri Debug          set_value: SupportedAudioStreamConfiguration to AQ4BAQICCQEBAQIBAAMBAQIBAA==
   HomeKitLink Siri Debug          set_value: SupportedRTPConfiguration to AgEA
   HomeKitLink Siri Debug          set_value: SupportedVideoStreamConfiguration to AbwBAQACDwMBAAEBAQIBAAIBAQIBAgMMAQJAAQICtAADAgoAAwwBAkABAgLwAAMCCgADDAEC4AECAg4BAwIKAAMMAQLgAQICaAEDAgoAAwwBAoACAgJoAQMCCgADDAECgAICAuABAwIKAAMMAQIABAICQAIDAgoAAwwBAgAEAgIAAwMCCgADDAECAAUCAtACAwIKAAMMAQIABQICwAMDAgoAAwwBAoAHAgI4BAMCCgADDAECQAYCArAEAwIKAA==
   HomeKitLink Siri Debug          set_value: SupportedAudioStreamConfiguration to AQ4BAQICCQEBAQIBAAMBAQIBAA==
   HomeKitLink Siri Debug          set_value: SupportedRTPConfiguration to AgEA
   HomeKitLink Siri Debug          set_value: SupportedVideoStreamConfiguration to AbwBAQACDwMBAAEBAQIBAAIBAQIBAgMMAQJAAQICtAADAgoAAwwBAkABAgLwAAMCCgADDAEC4AECAg4BAwIKAAMMAQLgAQICaAEDAgoAAwwBAoACAgJoAQMCCgADDAECgAICAuABAwIKAAMMAQIABAICQAIDAgoAAwwBAgAEAgIAAwMCCgADDAECAAUCAtACAwIKAAMMAQIABQICwAMDAgoAAwwBAoAHAgI4BAMCCgADDAECQAYCArAEAwIKAA==
   HomeKitLink Siri Debug          set_value: SupportedAudioStreamConfiguration to AQ4BAQICCQEBAQIBAAMBAQIBAA==
   HomeKitLink Siri Debug          useMotionSensor Enabled, setting up callbacks
   HomeKitLink Siri Debug          set_value: Manufacturer to org.cynic.indigo.securityspy
   HomeKitLink Siri Debug          set_value: Model to Camera
   HomeKitLink Siri Debug          set_value: SerialNumber to 303271220544621
   HomeKitLink Siri Debug          Adding Accessory:<accessory display_name='Sams Bed' services=['AccessoryInformation', 'Microphone', 'CameraRTPStreamManagement', 'CameraRTPStreamManagement', 'CameraRTPStreamManagement', 'MotionSensor']>
   HomeKitLink Siri Debug          set_value: FirmwareRevision to 0.2.22
   HomeKitLink Siri Debug          set_value: Manufacturer to Indigo 2022.1.0
   HomeKitLink Siri Debug          set_value: Model to HomeKitLink Bridge 303271
   HomeKitLink Siri Debug          set_value: SerialNumber to 3032711244
   HomeKitLink Siri Debug          Item :{'deviceid': 1866296567, 'devicename': 'Sams Bed', 'accessory': None, 'aid': 1866296567, 'subtype': 'SecuritySpyCamera', 'bridgeID': 303271, 'devicemodel': 'OutletLinc -2663-', 'devicesensor': '', 'manufacturername': ''} exisits for this bridge <accessory display_name='HomeKitLink Bridge 303271' services=['AccessoryInformation', 'HAPProtocolInformation']>
   HomeKitLink Siri Debug          Adding Accessory Name: Sams Bed,  ID  3032711866296567 , Type: SecuritySpyCamera
   HomeKitLink Siri Debug          sorting out camera options for ID:1866296567
   HomeKitLink Siri Debug          Config Now
{}

   HomeKitLink Siri Debug          Options:

{}

   HomeKitLink Siri Debug          set_value: Name to Sams Bed
   HomeKitLink Siri Debug          set_value: SerialNumber to default
   HomeKitLink Siri Debug          set_value: SupportedRTPConfiguration to AgEA
   HomeKitLink Siri Debug          set_value: SupportedVideoStreamConfiguration to AcoBAQACDwMBAAEBAQIBAAIBAQIBAgMMAQJAAQICtAADAg8AAwwBAkABAgLwAAMCDwADDAECQAECArQAAwIeAAMMAQJAAQIC8AADAh4AAwwBAuABAgIOAQMCHgADDAEC4AECAmgBAwIeAAMMAQKAAgICaAEDAh4AAwwBAoACAgLgAQMCHgADDAECAAQCAkACAwIeAAMMAQIABAICAAMDAh4AAwwBAgAFAgLQAgMCHgADDAECAAUCAsADAwIeAAMMAQKABwICOAQDAh4A
   HomeKitLink Siri Debug          set_value: SupportedAudioStreamConfiguration to AQ4BAQICCQEBAQIBAAMBAQIBAA==
   HomeKitLink Siri Debug          set_value: SupportedRTPConfiguration to AgEA
   HomeKitLink Siri Debug          set_value: SupportedVideoStreamConfiguration to AcoBAQACDwMBAAEBAQIBAAIBAQIBAgMMAQJAAQICtAADAg8AAwwBAkABAgLwAAMCDwADDAECQAECArQAAwIeAAMMAQJAAQIC8AADAh4AAwwBAuABAgIOAQMCHgADDAEC4AECAmgBAwIeAAMMAQKAAgICaAEDAh4AAwwBAoACAgLgAQMCHgADDAECAAQCAkACAwIeAAMMAQIABAICAAMDAh4AAwwBAgAFAgLQAgMCHgADDAECAAUCAsADAwIeAAMMAQKABwICOAQDAh4A
   HomeKitLink Siri Debug          set_value: SupportedAudioStreamConfiguration to AQ4BAQICCQEBAQIBAAMBAQIBAA==
   HomeKitLink Siri Debug          set_value: SupportedRTPConfiguration to AgEA
   HomeKitLink Siri Debug          set_value: SupportedVideoStreamConfiguration to AcoBAQACDwMBAAEBAQIBAAIBAQIBAgMMAQJAAQICtAADAg8AAwwBAkABAgLwAAMCDwADDAECQAECArQAAwIeAAMMAQJAAQIC8AADAh4AAwwBAuABAgIOAQMCHgADDAEC4AECAmgBAwIeAAMMAQKAAgICaAEDAh4AAwwBAoACAgLgAQMCHgADDAECAAQCAkACAwIeAAMMAQIABAICAAMDAh4AAwwBAgAFAgLQAgMCHgADDAECAAUCAsADAwIeAAMMAQKABwICOAQDAh4A
   HomeKitLink Siri Debug          set_value: SupportedAudioStreamConfiguration to AQ4BAQICCQEBAQIBAAMBAQIBAA==
   HomeKitLink Siri Error          Error in Get Bridge
Traceback (most recent call last):
  File "plugin.py", line 689, in get_bridge_multiple
    accessory = HKDevicesCameraSecuritySpy.SecuritySpyCamera(driver, plugin=self, indigodeviceid=item["deviceid"], display_name=item['devicename'], aid=deviceAID, config=cameraoptions)
  File "/Library/Application Support/Perceptive Automation/Indigo 2022.1/Plugins/HomeKitLink-Siri.indigoPlugin/Contents/Server Plugin/HKDevicesCameraSecuritySpy.py", line 254, in __init__
    if config["useMotionSensor"]:
KeyError: 'useMotionSensor'
   HomeKitLink Siri Error          Exception in single Bridge Startup
Traceback (most recent call last):
  File "plugin.py", line 920, in startsingleBridge
    self.driver_multiple[-1].add_accessory(accessory=self.get_bridge_multiple(self.driver_multiple[-1], self.bridge_multiple[-1], uniqueID))
  File "/Library/Application Support/Perceptive Automation/Indigo 2022.1/Plugins/HomeKitLink-Siri.indigoPlugin/Contents/Server Plugin/pyhap/accessory_driver.py", line 484, in add_accessory
    if accessory.aid is None:
AttributeError: 'NoneType' object has no attribute 'aid'
   HomeKitLink Siri Error          exception in device_start_comm(Siri): Python argument types in
    Device.updateStatesOnServer(Device)
did not match C++ signature:
    updateStatesOnServer(CDeviceBase {lvalue}, boost::python::api::object keyValueList, bool triggerEvents=True, bool clearErrorState=True)
   HomeKitLink Siri Debug          Still starting up. Ignoring device update
   HomeKitLink Siri Debug          Still starting up. Ignoring device update
   HomeKitLink Siri Debug          Still starting up. Ignoring device update
   HomeKitLink Siri                Camera Accessory Found, starting single thread to manage images.
   HomeKitLink Siri                Camera Thread is being spooled up to manage camera images
   HomeKitLink Siri Debug          Thread for managing all camera snapshots created and started
   HomeKitLink Siri Debug          Startup Refresh Snapshots images to Default for Camera: /Library/Application Support/Perceptive Automation/Indigo 2022.1/Preferences/Plugins/com.GlennNZ.indigoplugin.HomeKitLink-Siri/cameras/BabyCam.jpg
   HomeKitLink Siri                A total of 2 devices are currently selected to be published to HomeKit
   HomeKitLink Siri                0 Home Kit Accessories are now Started.
   HomeKitLink Siri                Given this difference, please review devices by Select Menu Item, Show Device Publications for more information.

May 9, 2022 at 8:37:56 AM
   HomeKitLink Siri Debug          Called getDeviceConfigUiValues(self, pluginProps, typeId, devId):
   HomeKitLink Siri Debug               (UiValuesDict : (dict)
     HomeKit_publishDevice_2 : true (bool)
     altName : Sams Bed (string)
     audioSelector : true (bool)
     bridgeUniqueID : 303271 (integer)
     deviceId : 220544621 (string)
     deviceSensor :  (string)
     deviceSubtype : SecuritySpyCamera (string)
     doorbellId :  (string)
     enablePublishFields : true (bool)
     isDeviceSelected : true (bool)
     selectdevice_or_AG : SS_camera (string)
     showALL : false (bool)
     showSubtypes : true (bool)
     tempSelector : false (bool), homeKitBridgeDevice, 515860483)
   HomeKitLink Siri Debug          Device list Generator:
 valuesDict UiValuesDict : (dict)
     HomeKit_publishDevice_2 : true (bool)
     altName : Sams Bed (string)
     audioSelector : true (bool)
     bridgeUniqueID : 303271 (integer)
     deviceId : 220544621 (string)
     deviceSensor :  (string)
     deviceSubtype : SecuritySpyCamera (string)
     doorbellId :  (string)
     enablePublishFields : true (bool)
     isDeviceSelected : true (bool)
     selectdevice_or_AG : SS_camera (string)
     showALL : false (bool)
     showSubtypes : true (bool)
     tempSelector : false (bool) 
 typeId homeKitBridgeDevice
   HomeKitLink Siri Debug          Device State Generator List:
[('actions', "actions == 'False'"), ('height', "height == '1520'"), ('motion', "motion == 'True'"), ('recording', "recording == 'False'"), ('sensitivity', "sensitivity == '84'"), ('state', "state == 'active'"), ('state.active', "state.active == 'True'"), ('state.disconnected', "state.disconnected == 'False'"), ('state.passive', "state.passive == 'False'"), ('state.preparing', "state.preparing == 'False'"), ('state.unavailable', "state.unavailable == 'False'"), ('type', "type == 'Network ONVIF'"), ('width', "width == '2688'")]
   HomeKitLink Siri Debug          subtype Generator/HomeKit Device Type. ValuesDict:UiValuesDict : (dict)
     HomeKit_publishDevice_2 : true (bool)
     altName : Sams Bed (string)
     audioSelector : true (bool)
     bridgeUniqueID : 303271 (integer)
     deviceId : 220544621 (string)
     deviceSensor :  (string)
     deviceSubtype : SecuritySpyCamera (string)
     doorbellId :  (string)
     enablePublishFields : true (bool)
     isDeviceSelected : true (bool)
     selectdevice_or_AG : SS_camera (string)
     showALL : false (bool)
     showSubtypes : true (bool)
     tempSelector : false (bool)
   HomeKitLink Siri Debug          Subtype set by user, returning that. Rightly or Wrongly.
   HomeKitLink Siri Debug          Show QR Code.  Saving Http page.
   HomeKitLink Siri Debug          Using path /Library/Application Support/Perceptive Automation/Indigo 2022.1/Plugins/HomeKitLink-Siri.indigoPlugin/Contents/Server Plugin/QRCode.html
   HomeKitLink Siri Debug          
   HomeKitLink Siri Debug          Device list Generator:
 valuesDict UiValuesDict : (dict)
     HomeKit_publishDevice_2 : true (bool)
     altName : Sams Bed (string)
     altNameLabel :  (string)
     audioSelector : true (bool)
     bridgeUniqueID : 303271 (integer)
     deviceId : 220544621 (string)
     deviceIdlabel :  (string)
     deviceSensor :  (string)
     deviceSensorLabel :  (string)
     deviceSubtype : SecuritySpyCamera (string)
     deviceSubtypeLabel :  (string)
     deviceSubtype_LightInfo :  (string)
     doorBellLabel :  (string)
     doorbellId :  (string)
     enablePublishFields : true (bool)
     isDeviceSelected : true (bool)
     publishDeviceLabel :  (string)
     qrcodeshow :  (string)
     save :  (string)
     saveDisabledLabel :  (string)
     saveLabel :  (string)
     saveLabelSpacing :  (string)
     saveWarningLabel :  (string)
     savelabelAudio :  (string)
     selectdevice_or_AG : SS_camera (string)
     sep1 :  (string)
     sep211a :  (string)
     sep211ab :  (string)
     sep21a1 :  (string)
     showALL : false (bool)
     showALLWarningLabel :  (string)
     showSubtypes : true (bool)
     tempSelector : false (bool) 
 typeId homeKitBridgeDevice
   HomeKitLink Siri Debug          Device State Generator List:
[('actions', "actions == 'False'"), ('height', "height == '1520'"), ('motion', "motion == 'True'"), ('recording', "recording == 'False'"), ('sensitivity', "sensitivity == '84'"), ('state', "state == 'active'"), ('state.active', "state.active == 'True'"), ('state.disconnected', "state.disconnected == 'False'"), ('state.passive', "state.passive == 'False'"), ('state.preparing', "state.preparing == 'False'"), ('state.unavailable', "state.unavailable == 'False'"), ('type', "type == 'Network ONVIF'"), ('width', "width == '2688'")]
   HomeKitLink Siri Debug          subtype Generator/HomeKit Device Type. ValuesDict:UiValuesDict : (dict)
     HomeKit_publishDevice_2 : true (bool)
     altName : Sams Bed (string)
     altNameLabel :  (string)
     audioSelector : true (bool)
     bridgeUniqueID : 303271 (integer)
     deviceId : 220544621 (string)
     deviceIdlabel :  (string)
     deviceSensor :  (string)
     deviceSensorLabel :  (string)
     deviceSubtype : SecuritySpyCamera (string)
     deviceSubtypeLabel :  (string)
     deviceSubtype_LightInfo :  (string)
     doorBellLabel :  (string)
     doorbellId :  (string)
     enablePublishFields : true (bool)
     isDeviceSelected : true (bool)
     publishDeviceLabel :  (string)
     qrcodeshow :  (string)
     save :  (string)
     saveDisabledLabel :  (string)
     saveLabel :  (string)
     saveLabelSpacing :  (string)
     saveWarningLabel :  (string)
     savelabelAudio :  (string)
     selectdevice_or_AG : SS_camera (string)
     sep1 :  (string)
     sep211a :  (string)
     sep211ab :  (string)
     sep21a1 :  (string)
     showALL : false (bool)
     showALLWarningLabel :  (string)
     showSubtypes : true (bool)
     tempSelector : false (bool)
   HomeKitLink Siri Debug          Subtype set by user, returning that. Rightly or Wrongly.
   HomeKitLink Siri Debug          ValidateDevice Config UI called UiValuesDict : (dict)
     HomeKit_publishDevice_2 : true (bool)
     altName : Sams Bed (string)
     altNameLabel :  (string)
     audioSelector : true (bool)
     bridgeUniqueID : 303271 (integer)
     deviceId : 220544621 (string)
     deviceIdlabel :  (string)
     deviceSensor :  (string)
     deviceSensorLabel :  (string)
     deviceSubtype : SecuritySpyCamera (string)
     deviceSubtypeLabel :  (string)
     deviceSubtype_LightInfo :  (string)
     doorBellLabel :  (string)
     doorbellId :  (string)
     enablePublishFields : true (bool)
     isDeviceSelected : true (bool)
     publishDeviceLabel :  (string)
     qrcodeshow :  (string)
     save :  (string)
     saveDisabledLabel :  (string)
     saveLabel :  (string)
     saveLabelSpacing :  (string)
     saveWarningLabel :  (string)
     savelabelAudio :  (string)
     selectdevice_or_AG : SS_camera (string)
     sep1 :  (string)
     sep211a :  (string)
     sep211ab :  (string)
     sep21a1 :  (string)
     showALL : false (bool)
     showALLWarningLabel :  (string)
     showSubtypes : true (bool)
     tempSelector : false (bool)
```

**Comment by Ghawken** [comment]
Thanks - that looks like a bug to me - I’ll fix!

**Comment by Ghawken** [comment]
Seems like you have tried to add deviceID: 1866296567 as a security spy camera, looks like Outlinc device of some description.   Then readded actual camera - which seems to set up okay.  

This non-camera is causing all the red/exceptions in the logs.

If you check the menu Item - Show Device Publications - may/may not work given how borked everything is.

But need to unpublished, and save this incorrect camera.

The Show ALL options obviously allows anything - but adding a light as a camera isn’t really going to work.  

I'll look at why it didn’t tell you this more clearly - as this is certainly one aspect that should not be any flexibility in...  (despite the general flexible approach when using showALL)

**Comment by Ghawken** [comment]
Should be fixed in version 0.2.31 - uploaded now.

**Comment by jandreaucodes** [comment]
Looks like that did the trick, thanks!
